tests/src/python/test_qgslayoutlegend.py

```diff
@@ -408,7 +404,6 @@ def testSymbolExpressions(self):
         label1 = legendnodes[0].evaluateLabel()
         label2 = legendnodes[1].evaluateLabel()
         label3 = legendnodes[2].evaluateLabel()
-        count.waitForFinished()
```

Comment:
Why these removals?

Reply:
There is already one waitForFinished before the first line modified, counterTask and count are the same object. I'm not sure why there was 2 waitForFinished calls in the first place (maybe to maximize the chance to actually wait that the task start?!), but the only one call is needed IMHO.

@roya0045 do you confirm?

Reply:
Yes, these were extra ways in an attempt to get the counter to finish before testing for the condition. If the waitforfinish actually works properly then these extra measures are not necessary anymore and I believe this should be a good implicit test for this fix.